internal/pkg/utils/utils.go

```diff
@@ -45,3 +46,42 @@ func FieldName(fa *ssa.FieldAddr) string {
 	}
 	return st.Field(fa.Field).Name()
 }
+
+// DecomposeType returns the path and name of a Named type
+// Returns empty strings if the type is not *types.Named
+func DecomposeType(t types.Type) (path, name string) {
```

Comment:
Similarly to `DecomposeFunction`, should this take a `types.Named` directly?

Reply:
Could, but bubbling up a type assertion is going to be messier than bubbling up a nil-check.  As it is, we take a `Type`, and we know that some `Type`s don't have names.  Either way, you can pass what you get to a matcher, and they are very unlikely to match anything, so it all shakes out.

Alternatively, something relatively simple here

```golang
		ssaType, ok := mem.(*ssa.Type)
		if !ok || !conf.IsSourceType(utils.DecomposeType(ssaType.Type())) {
			continue
		}
```

needs a second layer of assertion
```golang
		ssaType, ok := mem.(*ssa.Type)
		if !ok {
			continue
		}
		namedType, ok := ssaType.Type().(*types.Named)
		if ok && !conf.IsSourceType(utils.DecomposeType(namedType)) {
			continue
		}
```

Really, it comes down to the fact that the most common argument into `DecomposeType` is what comes out of `Dereference`, which is a `Type`.  We could change that to return only `Named`, or convert to some `DereferenceToUnderlyingNamed` or something, but that all feels beyond the scope of this PR.